#!/usr/bin/env python
PACKAGE = "rona_frontier"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add("framerate",  int_t, 0, "frame rate",    24,  15, 125)
#gen.add("gain",       int_t, 0, "gain",          0,   0,  255)

# parameters for weight estimation of a single frontier
# these parameters decide which frontier is the best for exploratioin
gen.add("dist_factor",         double_t, 0, "dist",               10, 0,  100)
gen.add("size_factor", 	       double_t, 0, "size_factor",          0, 0,  100)
gen.add("orientation_factor",  double_t, 0, "orientation",     		0, 0,  100)
gen.add("max_dist_threshold",  double_t, 0, "max_dist_threshold",  20, 0,  100)

exit(gen.generate(PACKAGE, "rona_frontier", "Frontier"))
